app/web/src/workers/locks.ts

```diff
@@ -28,19 +34,22 @@ export class ReadWriteLock {
   }
 
   async writeLock(callback: AnyFn): Promise<SqlValue[][]> {
-    return await navigator.locks.request(
+    const r = await navigator.locks.request(
       `${this.name}-reader`,
       { mode: "exclusive" },
       async () => {
         return await navigator.locks.request(
           `${this.name}-writer`,
           { mode: "exclusive" },
           async () => {
+            this.writeLockAcquired = true;
             return await callback();
           },
         );
       },
     );
+    this.writeLockAcquired = false;
```

Comment:
Should this be wrapped in a try so that the writeLockAcquired = false is executed in a finally? What happens if the callback throws an error?